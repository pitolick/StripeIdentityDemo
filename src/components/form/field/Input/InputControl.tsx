/**
 * 入力項目制御コンポーネント（react-hook-form）
 * @module InputControl
 */
import React, { forwardRef } from 'react'
import { Input, InputProps } from './Input'
import { InputBox, InputBoxProps } from '@/components/form/parts'

/**
 * 入力項目コンポーネントのプロパティ
 * @memo バリデーションスキーマ（zod）は各入力子コンポーネント内でexportしformコンポーネント内でresolverに指定する
 * @typedef {Object} InputControlProps
 */

export interface InputControlProps extends InputProps, Omit<InputBoxProps, 'children'> {}

/**
 * 入力項目制御コンポーネント（react-hook-form）
 * @param {InputControlProps} props - 入力項目コンポーネントのプロパティ
 * @returns {JSX.Element} - 入力項目コンポーネント
 */
export const InputControl = forwardRef<HTMLInputElement, InputControlProps>((props, ref) => {
  const { nameId, label, note, required = undefined, className, error, afterChildren, contentWidth, ...inputProps } = props
  const classNameProps = className ? className : ''
  // エラー時のclass名
  const errorClass = error ? `bg-red-pale outline outline-red outline-2 -outline-offset-2` : ''
  return (
    <InputBox nameId={nameId} label={label} note={note} required={required} error={error} afterChildren={afterChildren} contentWidth={contentWidth}>
      {/* 入力項目 */}
      <Input id={nameId} ref={ref} className={`${classNameProps} ${errorClass}`} {...inputProps} />
    </InputBox>
  )
})

InputControl.displayName = 'InputControl'
