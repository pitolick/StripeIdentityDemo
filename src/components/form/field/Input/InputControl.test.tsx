import React from 'react'
import { render } from '@testing-library/react'
import { InputControl } from './InputControl'

describe('InputControl', () => {
  it('正しくレンダリングされること', () => {
    const { container } = render(
      <InputControl
        nameId="inputName"
        label="名前"
        note={{ list: ['必須項目です'] }}
        required
        className="text-red"
        error={{ type: 'test' }}
        afterChildren={<span>After Children</span>}
        contentWidth="full"
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('デフォルト値が正しく設定されること', () => {
    const { container } = render(<InputControl nameId="inputName" />)
    expect(container).toMatchSnapshot()
  })
})
