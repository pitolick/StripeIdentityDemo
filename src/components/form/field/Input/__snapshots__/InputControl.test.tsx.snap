// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControl デフォルト値が正しく設定されること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <input
          class="  block w-full rounded-md border border-gray-500 px-3 py-2 text-sm file:mr-4 file:border-0 file:bg-gray-300 file:px-4 file:py-3 md:px-5 md:py-3 md:text-base"
          id="inputName"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`InputControl 正しくレンダリングされること 1`] = `
<div>
  <div
    class="w-full flex-auto space-y-2 false"
  >
    <div
      class="flex items-center justify-start gap-x-2"
    >
      <span
        class="text-xs text-center px-1.5 md:px-3 py-1 rounded-md whitespace-nowrap bg-red text-white"
      >
        必須
      </span>
      <label
        class="block text-sm font-bold"
        for="inputName"
      >
        名前
      </label>
    </div>
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <input
          class="text-red bg-red-pale outline outline-red outline-2 -outline-offset-2 block w-full rounded-md border border-gray-500 px-3 py-2 text-sm file:mr-4 file:border-0 file:bg-gray-300 file:px-4 file:py-3 md:px-5 md:py-3 md:text-base"
          id="inputName"
        />
        <span>
          After Children
        </span>
      </div>
      <ul>
        <li
          class="text-xs text-gray-700 before:me-1 ps-4 -indent-4 before:content-['※']"
        >
          必須項目です
        </li>
      </ul>
      <p
        class="text-sm text-red"
      />
    </div>
  </div>
</div>
`;
